dattri/task.py

```diff
@@ -353,20 +367,18 @@ def get_param(
         """
         self._load_checkpoints(index)
 
-        if layer_name:
+        if layer_name is not None:
             named_parameters = {
                 k: self.named_parameters[k]
                 for k in layer_name
                 if k in self.named_parameters
             }
-            # forced to layer_split
-            layer_split = True
```

Comment:
The parameter fetched by `task.get_param` will not be splited by parameter layers anymore, since that will affect the later hessian, hvp and other ihvp/ifvp calculation.